### ISSUE TITLE
Bug: The learning column is zeros.

### ISSUE THREAD

**Issue by cjdduarte** [glfharris/Syllabus#3]
The learning column is zeros.

![image](https://user-images.githubusercontent.com/42243532/46067382-12643600-c14d-11e8-8ad6-2804b1963813.png)

The configuration of the steps advances in more than 1 day. 
I believe it's another value in the database code:
(**QUEUE = 3**)

![image](https://user-images.githubusercontent.com/42243532/46067570-89013380-c14d-11e8-82cb-7c5ac666371b.png)

**Comment by cjdduarte** [comment]
This change solved:
```
def count_learning(deck=None, tag=None):
    cond = 'cards.queue == 1 or cards.queue == 3'
```

**Comment by glfharris** [comment]
Cheers, just incorporated!

**Comment by cjdduarte** [comment]
Sorry, but change has shown even the future lessons learned.

Learning due today = 53
Learning answered today and transferred to the future = 110

53+110=163

![image](https://user-images.githubusercontent.com/42243532/46077831-95df5080-c168-11e8-84aa-a973ace71e23.png)

The correct would be to show only 53,

I corrected by changing the code again for this:

`from anki.utils import intTime, ids2str`


```
def count_learning(deck=None, tag=None):
    cutoff = intTime() + mw.col.conf['collapseTime']
    today = mw.col.sched.today
    cond = '(cards.queue == 1  and due <= '+str(cutoff)+') or (cards.queue == 3 and due <= '+str(today)+')'
    return _card_note_query(condition=cond, deck=deck, tag=tag)
```

Now in the total column, OK

![image](https://user-images.githubusercontent.com/42243532/46078072-844a7880-c169-11e8-88ce-b6af92a11963.png)

But, I can not say why, it is accumulating (duplicating) in the lower lines.

![image](https://user-images.githubusercontent.com/42243532/46078157-be1b7f00-c169-11e8-9f56-deea2df0dd1c.png)


Can you correct that?

**Comment by cjdduarte** [comment]
An observation. I do not know what your concept is with the learning column.
depending, the number 163 (53+110) is correct (my example deck).

I separate what is due for the "present day" with the futures. If you make this my change, it will only show the ones due for the day and I saw that it does not have any column for the future. It's your decision.

in any case, does not solve the issue of the lower lines being duplicated. It's another problem.

**Comment by glfharris** [comment]
I guess it comes down to what people want from the learning column. If we say that:
queue == 1 is for the learning young cards
queue == 3 is for the learning mature cards
Then the way the current learning column works now kind of makes semantic sense (learning young + learning mature), ie showing the total number of learning cards.

I'm not quite sure what the accumulation is caused by, possibly the recursion used to generate counts for subdecks.

QuickEdit - Just seen your comment, I think at the moment I'll leave it displaying queue = 1 or 3. Along with the revlog table queries, I'm going to overhaul the general querying api which should hopefully make the accumulation you're seeing more understandable

**Comment by cjdduarte** [comment]
Right. Since your screen is statistic, I think it's just queue == 1 or queue == 3.

**Comment by glfharris** [comment]
Yeah, I think due today and learning today are more useful on the home window, like the addon you are showing does. Whereas I think the aim for this addon is to reason about the whole collection.

**Comment by cjdduarte** [comment]
I found the error. There were no parentheses. I forgot to have your query concatenate

```
def count_learning(deck=None, tag=None):
    cond = '(cards.queue == 1 or cards.queue == 3)'
    return _card_note_query(condition=cond, deck=deck, tag=tag)
```

**Comment by cjdduarte** [comment]
"or" without the parentheses killed the query. Sorry

**Comment by cjdduarte** [comment]
Its OK, now.

![image](https://user-images.githubusercontent.com/42243532/46081261-751bf880-c172-11e8-9154-b9e363586146.png)

**Comment by cjdduarte** [comment]
Let me know when to fix based on above, so you can test again.

**Comment by glfharris** [comment]
I believe the current version is up do date with:

> I guess it comes down to what people want from the learning column. If we say that:
> queue == 1 is for the learning young cards
> queue == 3 is for the learning mature cards
> Then the way the current learning column works now kind of makes semantic sense (learning young + learning mature), ie showing the total number of learning cards.
> 
> I'm not quite sure what the accumulation is caused by, possibly the recursion used to generate counts for subdecks.
> 
> QuickEdit - Just seen your comment, I think at the moment I'll leave it displaying queue = 1 or 3. Along with the revlog table queries, I'm going to overhaul the general querying api which should hopefully make the accumulation you're seeing more understandable

Are there still issues?

**Comment by cjdduarte** [comment]
Yes. I found out why the values are doubling in the lower columns.
There is missing parentheses "(" ")" in the "OR" condition.

Current code
```
def count_learning(deck=None, tag=None):
    cond = 'cards.queue == 1 or cards.queue == 3'
```
**Correction**:
```
def count_learning(deck=None, tag=None):
    cond = '(cards.queue == 1 or cards.queue == 3)'
```

**Comment by glfharris** [comment]
Huh, didn't know that would make a difference. I've included it in v0.1.3. Thanks for the troubleshooting!

**Comment by cjdduarte** [comment]
Now it's OK.
I thank you.

![image](https://user-images.githubusercontent.com/42243532/46109426-f1d0c600-c1b6-11e8-8781-f371ff830a55.png)
